dev/tasks/tasks.yml

```diff
@@ -984,14 +984,18 @@ tasks:
       flags: -e ARROW_SKYHOOK=ON
       image: ubuntu-cpp
 
-{% for debian_version in ["10", "11"] %}
-  test-debian-{{ debian_version }}-cpp:
+{% for arch, flags in [("amd64", ""), ("i386", "-e ARROW_S3=OFF -e ARROW_GANDIVA=OFF")] %}
```

Comment:
Nit, but for readability and maintainability I'd favour separate snippets for amd64 and i386.

Reply:
Sure, I can separate them.